### PR TITLE
Question: How do I pass in an existing object to a function?

### PR DESCRIPTION
**Pull request by ctaggart**
Hi Rob, this isn't meant to be merged. I'm just looking for some help with a little code review. Should it be possible to pass in an existing JavaScriptObject to a function call? So far, no luck. This is related to https://github.com/ctaggart/TsAst/issues/5 that I've been hacking on for a couple of days, trying to find a solution.

### CODE REVIEW

File: src/ConsoleHost/Program.cs [hunk 1]
```diff
@@ -18,20 +18,51 @@ static void Main(string[] args)
...
+
+                // How to I get the Number value?
+                var age = prsn.GetPropertyByName("age");
+                Console.WriteLine("age type: {0}", age.Type);
+                //var a = age as JavaScriptNum
+                //Console.WriteLine("age: {0}", age);
+
+                // Is it possible to pass a JavaScriptObject back in?
+                var getName = engine.EvaluateScriptText(@"(function(person){ return person.name; })();");
+                var n = getName.Invoke(new []{ person }); // JsErrorScriptException
```
Comment (ctaggart) [inline src/ConsoleHost/Program.cs]: I was hoping this would work, but it throws a `JsErrorScriptException`.

**Comment by robpaveza** [comment]
I don't think that you can do that the way you're trying.  JsParseText (which is what EvaluateScriptText calls down to) will create a new ambient, parameterless "function" that encapsulates your source, and that's what is returned.  

Try removing the `()` at the end of the script text you're evaluating.  That might evaluate to the function you're declaring rather than a function that executes the function.  :)

**Comment by ctaggart** [comment]
Thanks Rob! I able to pass the object successfully to the function now. Even with the `()`, it is still nested in an additional function that I wasn't expecting.

**Comment by robpaveza** [comment]
So, how did you get the inner function?

File: src/ConsoleHost/Program.cs [hunk 2]
```diff
@@ -18,20 +18,62 @@ static void Main(string[] args)
...
+                Console.WriteLine("age type: {0}", age.Type);
+                //var a = age as JavaScriptNum
+                //Console.WriteLine("age: {0}", age);
+
+                // Is it possible to pass a JavaScriptObject back in?
+                var getName = engine.EvaluateScriptText(@"(function(person){ return person.name; });");
+                Console.WriteLine("getName type: {0}", getName.Type); // Function
+
+                var getName2 = getName.Invoke(Enumerable.Empty<JavaScriptValue>()); // JsErrorScriptException
+                Console.WriteLine("getName2 type: {0}", getName2.Type); // Function
```
Comment (ctaggart) [inline src/ConsoleHost/Program.cs:54]: I got the function created by `EvaluateScriptText` then `Invoke`'d it to get the actual function. On my project, I just called `JsRunScript` [directly](https://github.com/ctaggart/TsAst/blob/0dd61989ae907988c56e7d6da2800820a76640f2/Program.fs#L107).

**Pull request closed without merging**